cmake/CMakeLists.txt

```diff
@@ -117,8 +117,9 @@ option(onnxruntime_EXTENDED_MINIMAL_BUILD "onnxruntime_MINIMAL_BUILD with suppor
 option(onnxruntime_MINIMAL_BUILD_CUSTOM_OPS "Add custom operator kernels support to a minimal build." OFF)
 option(onnxruntime_REDUCED_OPS_BUILD "Reduced set of kernels are registered in build via modification of the kernel registration source files." OFF)
 option(onnxruntime_DISABLE_EXTERNAL_INITIALIZERS "Don't allow models to load external data" OFF)
-cmake_dependent_option(onnxruntime_ENABLE_ORT_FORMAT_RUNTIME_GRAPH_OPTIMIZATION "Enable runtime graph optimization of ORT format models." ON
-                       "NOT onnxruntime_MINIMAL_BUILD OR onnxruntime_EXTENDED_MINIMAL_BUILD" OFF)
+cmake_dependent_option(onnxruntime_ENABLE_ORT_FORMAT_RUNTIME_GRAPH_OPTIMIZATION
+                       "Enable runtime graph optimization of ORT format models. Warning: Not yet ready for general use."
+                       OFF "NOT onnxruntime_MINIMAL_BUILD OR onnxruntime_EXTENDED_MINIMAL_BUILD" OFF)
 
```

Comment:
does the 'OR' get evaluated before the 'NOT'? can we use brackets to make it clearer if the NOT applies to just one or both of onnxruntime_MINIMAL_BUILD and onnxruntime_EXTENDED_MINIMAL_BUILD?

Reply:
> does the 'OR' get evaluated before the 'NOT'?

no, https://cmake.org/cmake/help/latest/command/if.html#condition-syntax says NOT is evaluated before OR

> can we use brackets to make it clearer if the NOT applies to just one or both of onnxruntime_MINIMAL_BUILD and onnxruntime_EXTENDED_MINIMAL_BUILD?

I think that would work as of CMake 3.22. earlier, there are some limitations mentioned here about parentheses groups:
https://cmake.org/cmake/help/latest/policy/CMP0127.html#policy:CMP0127

so conservatively I'll leave it not parenthesized